backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java

```diff
@@ -11,13 +14,20 @@
 @RequiredArgsConstructor
 public class ConversationService {
     private final LLMService llmService;
+    private final ConversationPersistenceManager conversationPersistenceManager;
 
     public Conversation start() {
-        return null;
+        Conversation conversation =
+                Conversation.builder().text(null).discussions(null).build();
+        return conversationPersistenceManager.save(conversation);
     }
 
-    public Conversation getByID(UUID id) throws ConversationNotFoundException {
-        return null;
+    public List<Conversation> getAll() {
+        return conversationPersistenceManager.findAll();
+    }
+
+    public Optional<Conversation> getByID(UUID id) {
```

Comment:
shouldn't this throw an error if the id is not found ?

Reply:
No, this returns optional Conversation. We check if it is empty in the controller, and then exception handler takes care of it.